File: Data Structures & Algorithms/logger-rate-limiter/submission-0.java
class Logger {

    private final static int DELAY = 10;
    private Map<String, Integer> map;
    public Logger() {
        map = new HashMap<>();
    }
    
    public boolean shouldPrintMessage(int timestamp, String message) {
        if (map.get(message) == null || timestamp >= map.get(message)) {
            map.put(message, timestamp+DELAY);
            return true;
        }
        return false;
    }
}

/**
 * Your Logger object will be instantiated and called as such:
 * Logger obj = new Logger();
 * boolean param_1 = obj.shouldPrintMessage(timestamp,message);
 */

File: Data Structures & Algorithms/implement-stack-using-queues/submission-0.ts
class MyStack {

    private list: number[] = []
    constructor() {}

    /**
     * @param {number} x
     * @return {void}
     */
    push(x: number): void {
        this.list.push(x);
    }

    /**
     * @return {number}
     */
    pop(): number {
        return this.list.pop();
    }

    /**
     * @return {number}
     */
    top(): number {
        return this.list[this.list.length-1];
    }

    /**
     * @return {boolean}
     */
    empty(): boolean {
        return this.list.length === 0;
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */
